hydromt/data_adapter/rasterdataset.py

```diff
@@ -622,3 +625,29 @@ def _parse_zoom_level(
         zl = zls[-1] if all(smaller) else zls[max(smaller.index(False) - 1, 0)]
         logger.info(f"Getting data for zoom_level {zl} based on res {zoom_level}")
         return zl
+
+    def detect_spatial_range(self, ds=None) -> geometry:
+        """Detect spatial range."""
+        if ds is None:
+            ds = self.get_data()
+        return box(*ds.raster.bounds)
```

Comment:
note to self: always convert to WGS84 (EPSG 4326) (follow STAC) just return numbers

Reply:
Update on note: STAC makes no reference to CRS, so presumably they assume the provided extents are always in the same CRS as the data, however, I think we should still have everything in WGS84 to make searching easier. We can always do a transform when converting to STAC

Reply:
@DirkEilander Thoughts?

Reply:
Most seem to be WGS84 indeed. Pretty surprising indeed. If this extension is supported then of course getting the CRS can be done by looking at collection / item metadata.

Reply:
Example to load into xarray: https://github.com/opendatacube/odc-stac

Reply:
What do we do/return if the CRS is not set?

Reply:
You can:
- try to read some description and parse for CRS description
- try WGS84

Any other options come to mind?
Perhaps some of the files already contain the CRS information (headers in zarr / COG)

Reply:
CRS is required for any dataset in HydroMT. If it is not set in the catalog it should be set in the data source itself.

Reply:
Then we should be okay. I implemented the separate functions Dirk suggested, so I think we should be okay now